ote_sdk/ote_sdk/entities/model_template.py

```diff
@@ -75,28 +75,139 @@ def __str__(self) -> str:
+        value: int,
+        task_info: TaskInfo,
+    ):
+        """
+        :param value: Unique integer for .value property of Enum (auto() does not work)
+        :param task_info: NamedTuple containing information about the task's capabilities
+        """
+        self._value_ = value
+        self.domain = task_info.domain
+        self.is_trainable = task_info.is_trainable
+        self.is_anomaly = task_info.is_anomaly
+        self.is_global = task_info.is_global
+
+    NULL = 1, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    DATASET = 2, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    CLASSIFICATION = 3, TaskInfo(
+        domain=Domain.CLASSIFICATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=True,
+        is_local=False,
+    )
+    SEGMENTATION = 4, TaskInfo(
+        domain=Domain.SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    DETECTION = 5, TaskInfo(
+        domain=Domain.DETECTION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    ANOMALY_DETECTION = 6, TaskInfo(
+        domain=Domain.ANOMALY_DETECTION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=False,
+        is_local=True,
+    )
+    CROP = 7, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    TILE = 8, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    INSTANCE_SEGMENTATION = 9, TaskInfo(
+        domain=Domain.INSTANCE_SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    ACTIVELEARNING = 10, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    ANOMALY_SEGMENTATION = 11, TaskInfo(
+        domain=Domain.ANOMALY_SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=False,
+        is_local=True,
+    )
+    ANOMALY_CLASSIFICATION = 12, TaskInfo(
+        domain=Domain.ANOMALY_CLASSIFICATION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=True,
+        is_local=False,
+    )
+    ROTATED_DETECTION = 13, TaskInfo(
+        domain=Domain.ROTATED_DETECTION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
 
     def __str__(self) -> str:
         return str(self.name)
 
+    def __repr__(self) -> str:
+        return self.name
```

Comment:
What's the difference to the `__str__` here?

Reply:
If only `__str__` is defined and you print a list of TaskType, then it will display like `[<TaskType.NULL: 1>, <TaskType.CLASSIFICATION: 2>` which is ugly. This is because the containers use `__repr__` of the contained objects to represent them. Conversely, if you define only `__repr__`, then `print(TaskType.DETECTION)` will output `TaskType.DETECTION` because it uses the `__str__` inherited from the `Enum` class.
Redefining both methods will produce the most human-readable output in any case.

Reply:
Yeah, thanks. :)

I was trying to figure out, what's the difference in the implementation of `__str__` and `__repr__` in this particular case:

```python
    def __str__(self) -> str:
        return str(self.name)

    def __repr__(self) -> str:
        return self.name
```

That is, when `str(self.name)` differs from `self.name`?

Reply:
I think it's just redundant, `self.name` is already a string, it can be removed